common/src/main/java/org/apache/cassandra/sidecar/common/NodeSettings.java

```diff
@@ -16,33 +16,80 @@
  * limitations under the License.
  */
 
-
 package org.apache.cassandra.sidecar.common;
 
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Map;
 import java.util.Objects;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Holds information about the specific node settings
  */
 public class NodeSettings
 {
+    private static final Logger LOGGER = LoggerFactory.getLogger(NodeSettings.class);
+    private static final String VERSION = "version";
+    private static final Map<String, String> SIDECAR = Collections.singletonMap(VERSION, getSidecarVersion());
```

Comment:
I don't recommend opening a file during static initialization of variables. While this seems like a good idea, it should be avoided. I would prefer reading this explicitly during server initialization and populating the node settings (perhaps injecting it via Guice?).

Reply:
In this specific case it was reading from an internal resource, and swallowing all possible exceptions, but I agree that this can lead to issues that are difficult to identify. Added a separate commit that uses Guice to retrieve current Sidecar version.